src/Symfony/Component/Notifier/Bridge/Twitter/TwitterTransport.php

```diff
@@ -0,0 +1,297 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Twitter;
+
+use Symfony\Component\Mime\Part\DataPart;
+use Symfony\Component\Mime\Part\Multipart\FormDataPart;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\ChunkInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class TwitterTransport extends AbstractTransport
+{
+    protected const HOST = 'api.twitter.com';
+
+    private static $nonce;
+
+    private string $apiKey;
+    private string $apiSecret;
+    private string $accessToken;
+    private string $accessSecret;
+
+    public function __construct(#[\SensitiveParameter] string $apiKey, #[\SensitiveParameter] string $apiSecret, #[\SensitiveParameter] string $accessToken, #[\SensitiveParameter] string $accessSecret, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        parent::__construct($client, $dispatcher);
+
+        $this->apiKey = $apiKey;
+        $this->apiSecret = $apiSecret;
+        $this->accessToken = $accessToken;
+        $this->accessSecret = $accessSecret;
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('twitter://%s', $this->getEndpoint());
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage && (null === $message->getOptions() || $message->getOptions() instanceof TwitterOptions);
+    }
+
+    public function request(string $method, string $url, array $options): ResponseInterface
+    {
+        $url = 'https://'.str_replace('api.', str_starts_with($url, '/1.1/media/') ? 'upload.' : 'api.', $this->getEndpoint()).$url;
+
+        foreach (\is_array($options['body'] ?? null) ? $options['body'] : [] as $v) {
+            if (!$v instanceof DataPart) {
+                continue;
+            }
+            $formDataPart = new FormDataPart($options['body']);
+            foreach ($formDataPart->getPreparedHeaders()->all() as $header) {
+                $options['headers'][] = $header->toString();
+            }
+            $options['body'] = $formDataPart->bodyToIterable();
+            break;
```

Comment:
```suggestion
            if (!$v instanceof DataPart) {
                continue;
            }

            $formDataPart = new FormDataPart($options['body']);

            foreach ($formDataPart->getPreparedHeaders()->all() as $header) {
                $options['headers'][] = $header->toString();
            }

            $options['body'] = $formDataPart->bodyToIterable();
            break;
```

This is just personal opinion, but for me adding those "extra" empty lines makes this part easier to read.